### PR TITLE
Added ChangeLog entity state check Integration Test

### PR DESCRIPTION
**Pull request by WASandaruwan**
Added ChangeLog entity state check Integration Test

### CODE REVIEW

File: tests/OnlineSales.Tests/Environment/TestApplication.cs [hunk 1]
```diff
@@ -44,6 +45,19 @@ public void PopulateBulkData(dynamic bulkItems)
         }
     }
 
+    public T? Get<T>(Expression<Func<T, bool>> predicate)
```
Comment (peterliapin) [inline tests/OnlineSales.Tests/Environment/TestApplication.cs:48]: @WASandaruwan, this is too specific function to be placed into TestApplication class. What if the next time I want to request a list of records? Or I want some kind of join... It is better to return only the DBContext instance and the the client code can do whatever it need to do with the DBContext.

File: tests/OnlineSales.Tests/SimpleTableTests.cs [hunk 2]
```diff
@@ -48,17 +50,39 @@ public virtual async Task UpdateItemNotFoundTest()
...
-        await PatchTest(testCreateItem.Item2, testUpdateItem!);
+        var result = Get<ChangeLog>(c => c.ObjectId == item!.Id && c.ObjectType == typeof(T).Name && c.EntityState == EntityState.Added);
 
-        var item = await GetTest<T>(testCreateItem.Item2);
+        result.Should().NotBeNull();
+    }
 
-        item.Should().BeEquivalentTo(testCreateItem.Item1);
+    [Fact]
+    public async Task CreateAndUpdateItemTest()
```
Comment (peterliapin) [inline tests/OnlineSales.Tests/SimpleTableTests.cs:65]: How is it related with ChangeLog testing?

File: tests/OnlineSales.Tests/SimpleTableTests.cs [hunk 3]
```diff
@@ -48,17 +50,39 @@ public virtual async Task UpdateItemNotFoundTest()
...
+    {
+        var createAndUpdateItems = await CreateAndUpdateItem();
+
+        var item = await GetTest<T>(createAndUpdateItems.testCreateItem.Item2);
+
+        var result = Get<ChangeLog>(c => c.ObjectId == item!.Id && c.ObjectType == typeof(T).Name && c.EntityState == EntityState.Modified);
+
+        result.Should().NotBeNull();
+
+        result!.EntityState.Should().Equals(EntityState.Added);
```
Comment (peterliapin) [inline tests/OnlineSales.Tests/SimpleTableTests.cs:85]: This test will never be able to pass as you find records with EntityState.Modified status and here you check it for EntityState.Added. Am I missing something?

File: tests/OnlineSales.Tests/SimpleTableTests.cs [hunk 4]
```diff
@@ -149,6 +187,12 @@ protected virtual void GenerateBulkRecords(int dataCount, Action<TC>? populateAt
         App.PopulateBulkData(bulkEntitiesList);
     }
 
+    protected virtual TW? Get<TW>(Expression<Func<TW, bool>> predicate)
```
Comment (peterliapin) [inline tests/OnlineSales.Tests/SimpleTableTests.cs:190]: What is the purpose of this method? Why not to use App.Get directly?

**Review by peterliapin — CHANGES_REQUESTED** [review]

File: tests/OnlineSales.Tests/SimpleTableTests.cs (hunk 2 above)
Comment (WASandaruwan) [reply tests/OnlineSales.Tests/SimpleTableTests.cs:65]: I think here your concern is Why Did I Change the CreateAndUpdateItemTest? I made a single method as "CreateAndUpdateItem" and referred to it in both places (Change long one and CreateAndUpdateItemTest)

**Pull request merged**